src/components/grid/GridContainer.tsx

```diff
@@ -169,12 +76,16 @@ const GridContainer: FC<GridContainerProps> = (props: GridContainerProps) => {
 
 const dispatchProps = {
   setAlerts,
+  setWidgetConfigs,
 };
+
 const mapStateToProps = (state: RootState) => ({
   assetId: state.appState.asset?.id,
   user: state.authState.userInfo?.name,
+  localWidgetConfigs: state.appState.localWidgetConfigs,
```

Comment:
Things would be much cleaner if you kept layout information separately from widget configs. Currently widget configs is changed from multiple locations - through configurators and through grid.